src/test/java/com/puppycrawl/tools/checkstyle/AbstractModuleTestSupport.java

```diff
@@ -370,6 +376,19 @@ private static void verifyConfig(Configuration testConfig, Configuration parsedC
                     .that(inputConfiguration.getDefaultPropertyValue(property))
                     .isEqualTo(testConfig.getAttribute(property));
         }
+        final Set<Integer> expectedViolationLines = new TreeSet<>();
```

Comment:
I used TreeSet to maintain a sorted order of unique line numbers.
I feel, in the interest of BDD, we should add functionality to support comments like ```// violation 2x``` to verify if the violations occur the correct number of times. This could be a new issue under #10076. If mentors agree, I'll move this to a comment in the parent issue for discussion purpose.

Reply:
> // violation 2x

I do not see reason in support of this. In do not see how it is possible to have 2 different violations on the same line from the same Check. Lets postpone this until we see real use case.

Reply:
For example, WhitespaceAfterCheck throws multiple violations on the same line. :)
https://github.com/checkstyle/checkstyle/blob/0d9b3afe600b247eaccf758a0c6ae2bc8409a268/src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheckTest.java#L87-L91

Reply:
Ok, same line, but not the same column.

Ok, let's keep it.

Reply:
> Ok, let's keep it.

@romani Issue Opened: https://github.com/checkstyle/checkstyle/issues/10190
@timurt please approve, add to project and update description in parent issue

I'll work on it after this is merged.

Reply:
It should be postponed a bit. I need to see general bdd concept to work.
Not simple cases might stay in old way of testing.

Reply:
https://github.com/checkstyle/checkstyle/issues/10190#issuecomment-869103261

Reply:
lets try another approach, during comparing real configuration with the parsed one, you should select unique line numbers for violations from real config and compare with parsed violation list